@ecomplus/widget-fb-pixel/src/lib/watch-app-routes.js

```diff
@@ -60,11 +60,24 @@ export default (fbq, options) => {
         } else {
           eventID = orderId
         }
+        let buyerSha256
+        if (Array.isArray(order.buyers) && order.buyers.length) {
+          const buyerId = order.buyers[0]._id
+          async function sha256Text(message) {
+            const msgUint8 = new TextEncoder().encode(message);                         
+            const hashBuffer = await crypto.subtle.digest('SHA-256', msgUint8);         
+            const hashArray = Array.from(new Uint8Array(hashBuffer));
+            const hashHex = hashArray.map((b) => b.toString(16).padStart(2, '0')).join('');
+            return hashHex;
+          }
+          buyerSha256 = await sha256Text(buyerId)
+        }
         if (options.disablePurchase !== true) {
           fbq('Purchase', {
             ...getPurchaseData(order),
             order_id: orderId,
-            eventID
+            eventID,
+            external_id: buyerSha256
```

Comment:
Além de não me parecer necessário (a solução pra deduplicar purchase é desativar o purchase no widget), qual referência você pegou? Os parâmetros do FB não costumam ser snake case, o `order_id` é um custom, batendo o olho eu acho que isso aqui tá errado.

Reply:
https://developers.facebook.com/docs/marketing-api/conversions-api/parameters/customer-information-parameters#

Reply:
Ele tá mandando no init nesse exemplo, funciona no evento também?

> Hashing recommended.
Any unique ID from the advertiser, such as loyalty membership IDs, user IDs, and external cookie IDs. You can send one or more external IDs for a given event.
If an External ID is being sent via other channels, it should be in the same format as when sent via the [Conversions API](https://developers.facebook.com/docs/marketing-api/conversions-api/parameters/external-id).

_it should be in the same format_: não tem hash quando enviado server side, nem deveria ter na verdade porque (diferente do e-mail se enviado client side, por exemplo) não é uma informação crítica. Acredito que a recomendação deles aí é pro caso de ser um dado privado da aplicação.

Reply:
Ou seja, larga mão desse crypto, async/await e afins, manda só o buyer ID (raw) mesmo, e de qualquer forma confirma depois se isso pode ser enviado no purchase mesmo, já que é um parâmetro do usuário (do init) e não exatamente da conversão

Reply:
Foi mal, aperreado com bling aqui, enviando varios pedidos via codigo no meu pc. Esse external id pode ser enviado no purchase, porque é um erro que mostra lá no business center do pixel, informando justamente a ausencia desse parâmetro no evento purchase.

Reply:
> _it should be in the same format_: não tem hash quando enviado server side, nem deveria ter na verdade porque (diferente do e-mail se enviado client side, por exemplo) não é uma informação crítica. Acredito que a recomendação deles aí é pro caso de ser um dado privado da aplicação.

Achava que enviava criptografado, mas acabei de pesquisar no código lá no sdk e realmente, não tem cripografia nos dados - https://github.com/facebook/facebook-nodejs-business-sdk/blob/main/src/objects/businessdataapi/user-data.js#L327-L329